token/program-2022/src/extension/mod.rs

```diff
@@ -160,26 +150,31 @@ fn type_and_tlv_indices<S: BaseState>(rest_input: &[u8]) -> Result<(usize, usize
 pub struct StateWithExtensions<'data, S: BaseState> {
     /// Unpacked base data
     pub base: S,
-    /// Unpacked account type
-    pub account_type: AccountType,
```

Comment:
This field seems unnecessary.  The desired `BaseState` is validated on `unpack()` so the caller already knows the account type

Reply:
Yep, definitely